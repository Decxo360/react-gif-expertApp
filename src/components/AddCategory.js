import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({handleAdd}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = ({target}) => {
    setInputValue(target.value);
  }
  
  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return; 
    // setCategory(category => [inputValue,...category]);
    setInputValue("")
    handleAdd(inputValue.trim())
  }

  return (
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
  )
}

// AddCategory.propTypes = {
//   setCategory: PropTypes.func.isRequired
// }