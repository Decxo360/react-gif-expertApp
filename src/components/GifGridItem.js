import React from 'react'
import PropTypes from "prop-types"

export const GifGridItem = ({title,url}) => {
  return (
    <div className='card animate__animated animate__bounce'>
        <img src={url} alt={title}></img>
        <h3>{title}</h3>
    </div>
  )
}
GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}