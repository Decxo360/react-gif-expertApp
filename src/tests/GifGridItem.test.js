import { shallow } from "enzyme";
import { GifGridItem } from "../components/GifGridItem";


describe('Se esta trabajando en GifGridItem', () => {
    const title = "Este es el titulo";
    const url = "https://www.youtube.com"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Tiene que imprimir el componente GifgridItem', () => {

        expect(wrapper).toMatchSnapshot();

    })
})