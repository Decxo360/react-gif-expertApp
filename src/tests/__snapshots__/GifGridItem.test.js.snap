// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se esta trabajando en GifGridItem Tiene que imprimir el componente GifgridItem 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Este es el titulo"
    src="https://www.youtube.com"
  />
  <h3>
    Este es el titulo
  </h3>
</div>
`;
